src/Microsoft.Azure.WebJobs.Host/Bindings/QueueBindingRules.cs

```diff
@@ -0,0 +1,217 @@
+            converterManager.AddConverter<IStorageQueueMessage, string>(ConvertCloudQueueMessage2String);
+            converterManager.AddConverter<IStorageQueueMessage, byte[]>(ConvertCloudQueueMessage2ByteArray);
+
+            converterManager.AddConverter<CloudQueueMessage, IStorageQueueMessage>(ConvertReal2CloudQueueMessage);
+
+            var bf = new BindingFactory(nameResolver, converterManager);
+
+            var ruleQueueOutput = bf.BindToAsyncCollector<QueueAttribute, IStorageQueueMessage>(BuildFromQueueAttribute, ToParameterDescriptorForCollector, FixerUpper);
+            var ruleQueueClient = bf.BindToExactAsyncType<QueueAttribute, IStorageQueue>(BuildClientFromQueueAttributeAsync, ToParameterDescriptorForCollector, FixerUpper);
+            var ruleQueueClient2 = bf.BindToExactAsyncType<QueueAttribute, CloudQueue>(BuildRealClientFromQueueAttributeAsync, ToParameterDescriptorForCollector, FixerUpper);
+            var queueRules = new GenericCompositeBindingProvider<QueueAttribute>(
+                (attr) => ValidateQueueAttribute(attr, nameResolver),
+                ruleQueueClient, ruleQueueClient2, ruleQueueOutput);
+
+            return queueRules;
+        }
+
+        // Hook to apply hacky rules all at once. 
+        private async Task<QueueAttribute> FixerUpper(QueueAttribute attrResolved, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            // Look for [Storage] attribute and squerrel aover 
+            IStorageAccount account = await _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver);
+            StorageClientFactoryContext clientFactoryContext = new StorageClientFactoryContext
+            {
+                Parameter = parameter
+            };
+            IStorageQueueClient client = account.CreateQueueClient(clientFactoryContext);
+
+            // Create the queue if needed?
+            var queue = client.GetQueueReference(attrResolved.QueueName);
+            bool create = false;
+            if (create)
+            {
+                await queue.CreateIfNotExistsAsync(CancellationToken.None);
+            }
+
+            // Normalize and validate? 
+            string queueName = attrResolved.QueueName.ToLowerInvariant(); // must be lowercase. coerce here to be nice.
+            QueueClient.ValidateQueueName(queueName);
+
+            // Lowercase
+            return new ResolvedQueueAttribute(queueName, client);
+        }
+
+        private ParameterDescriptor ToParameterDescriptorForCollector(QueueAttribute attr, ParameterInfo parameter, INameResolver nameResolver)
+        {
+            Task<IStorageAccount> t = Task.Run(() =>
+                _accountProvider.GetStorageAccountAsync(parameter, CancellationToken.None, nameResolver));
+            string accountName = t.GetAwaiter().GetResult().Credentials.AccountName;
+
+            return new QueueParameterDescriptor
```

Comment:
Yeah, the Dashboard tests try all the binding permutations and verify the expected output. We want to make sure we don't break those. Tests are here: https://github.com/Azure/azure-webjobs-sdk-dashboard-tests/blob/master/test/Dashboard.EndToEndTests/Tests/ArgumentsDisplay/QueueArgumentsDisplayTests.cs

Are you able to run them locally? If not, it might be that since we've all moved to the latest VS the tests no longer run for us. I can update the test proj to try to get it running for VS2015.